src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPalExpress.class.php

```diff
@@ -117,32 +117,85 @@ public function PostProcessExternalPaymentHandlerHook()
+        $sCompany = $this->aCheckoutDetails['BUSINESS'] ?? '';
+        $sFirstname = $this->aCheckoutDetails['FIRSTNAME'] ?? '';
+        $sLastname = $this->aCheckoutDetails['LASTNAME'] ?? '';
+        $sStreet = $this->aCheckoutDetails['SHIPTOSTREET'] ?? '';
+        $sCity = $this->aCheckoutDetails['SHIPTOCITY'] ?? '';
+        $sPostalCode = $this->aCheckoutDetails['SHIPTOZIP'] ?? '';
+        $sTelefon = $this->aCheckoutDetails['PHONENUM'] ?? '';
+        $addressAdditionalInfo = $this->aCheckoutDetails['SHIPTOSTREET2'] ?? '';
+
+        $aBilling = [
+            'name' => $sMail,
+            'email' => $sMail,
+            'company' => $sCompany,
+            'data_extranet_salutation_id' => '',
+            'firstname' => $sFirstname,
+            'lastname' => $sLastname,
+            'street' => $sStreet,
+            'streenr' => '',
+            'city' => $sCity,
+            'postalcode' => $sPostalCode,
+            'telefon' => $sTelefon,
+            'fax' => '',
+            'data_country_id' => $oShippingCountry->id ?? '',
+            'address_additional_info' => $addressAdditionalInfo,
+        ];
+
+        $shippingFirstAnLastName = $this->getPayPalResponseShippingFirstAnLastName(
+            $sFirstname,
+            $sLastname,
+            $this->aCheckoutDetails['SHIPTONAME'] ?? null
+        );
+        $aShipping = [
+            'company' => $sCompany,
+            'data_extranet_salutation_id' => '',
+            'firstname' => $shippingFirstAnLastName['firstName'],
+            'lastname' => $shippingFirstAnLastName['lastName'],
+            'street' => $sStreet,
+            'streenr' => '',
+            'city' => $sCity,
+            'postalcode' => $sPostalCode,
+            'telefon' => $sTelefon,
+            'fax' => '',
+            'data_country_id' => $oShippingCountry->id ?? '',
+            'address_additional_info' => $addressAdditionalInfo,
+        ];
 
-        $aBilling = array('name' => $sMail, 'email' => $sMail, 'company' => $sCompany, 'data_extranet_salutation_id' => '', 'firstname' => $sFirstname, 'lastname' => $sLastname, 'street' => $sStreet, 'streenr' => '', 'city' => $sCity, 'postalcode' => $sPostalcode, 'telefon' => $sTelefon, 'fax' => '', 'data_country_id' => $oShippingCountry->id, 'address_additional_info' => $addressAdditionalInfo);
+        $this->postProcessBillingAndShippingAddress($aBilling, $aShipping);
+    }
 
-        $sShippingLastName = (array_key_exists('SHIPTONAME', $this->aCheckoutDetails) && '' != $this->aCheckoutDetails['SHIPTONAME']) ? $this->aCheckoutDetails['SHIPTONAME'] : $sFirstname.' '.$sLastname;
+    /**
+     * use buyer's first and lastname, if no "shipToName" is provided,
+     * or if "shipToName" is just a simple creation out of both.
+     */
+    private function getPayPalResponseShippingFirstAnLastName(
+        string $userFirstName,
+        string $userLastName,
+        ?string $shipToName
+    ): array {
```

Comment:
Das anwenden von Codestyle-formattierung nach unseren Regeln (PSR/Symfony)  verweigert das und setzt `): array {` in eine Zeile...

Reply:
What? 
https://symfony.com/doc/current/contributing/code/standards.html

I think you are wrong. Code style for methods was and is { in next row. Which makes sense if you have a return type defined.
If statements need the { in the same line, but not methods.

Reply:
Your link sadly proves nothing. I don't think I am wrong, because I am using the symfony2 or psr coding style rules applied by phpstorm, we accepted on.

Reply:
There is not a single class in the core with braces not at the next line. 
We should check your coding standard rule set. Please change the code to braces in next lines. We can discuss the issue in person.